pkg/query-service/app/logs/v3/enrich_query_test.go

```diff
@@ -289,3 +289,154 @@ func TestEnrichParams(t *testing.T) {
 		})
 	}
 }
+
+var testJSONFilterEnrichData = []struct {
+	Name   string
+	Filter v3.FilterItem
+	Result v3.FilterItem
+}{
+	{
+		Name: "array string",
+		Filter: v3.FilterItem{
+			Key: v3.AttributeKey{
+				Key:      "body.requestor_list[*]",
+				DataType: v3.AttributeKeyDataTypeUnspecified,
+				Type:     v3.AttributeKeyTypeUnspecified,
+			},
+			Operator: "has",
+			Value:    "index_service",
+		},
+		Result: v3.FilterItem{
+			Key: v3.AttributeKey{
+				Key:      "body.requestor_list[*]",
+				DataType: v3.AttributeKeyDataTypeArrayString,
+				Type:     v3.AttributeKeyTypeUnspecified,
+				IsJSON:   true,
+			},
+			Operator: "has",
+			Value:    "index_service",
+		},
+	},
+	{
+		Name: "int64",
+		Filter: v3.FilterItem{
+			Key: v3.AttributeKey{
+				Key:      "body.intx",
+				DataType: v3.AttributeKeyDataTypeUnspecified,
+				Type:     v3.AttributeKeyTypeUnspecified,
+			},
+			Operator: "=",
+			Value:    10,
```

Comment:
Most values you get from the frontend will have string type.

```suggestion
			Value:    "10",
```

Reply:
yeah test for that is present in  `TestParseStrValue` . These tests were added before I considered the string to value conversion thing. So kept them.